pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -363,13 +355,15 @@ func (exp *flowExporter) sendTemplateSet(templateSet ipfixentities.Set, isIPv6 b
 		ieWithValue := ipfixentities.NewInfoElementWithValue(element, nil)
 		elements = append(elements, ieWithValue)
 	}
-
-	err := templateSet.AddRecord(elements, templateID)
+	exp.ipfixSet.ResetSet()
```

Comment:
the only difference is that we call `ResetSet` first? Not really clear from the PR description given that we were already calling `PrepareSet`:

> We need to prepare the set as a template set before sending the template record

I thought calling `ResetSet` was the key to the bug fix, but there is no mention of it in the PR description.

Reply:
Revised PR description.